Library/Homebrew/cli/args.rb

```diff
@@ -5,16 +5,23 @@
 module Homebrew
   module CLI
     class Args < OpenStruct
-      attr_accessor :processed_options
+      attr_reader :processed_options, :args_parsed
       # undefine tap to allow --tap argument
       undef tap
 
       def initialize(argv:)
         super
         @argv = argv
+        @args_parsed = false
         @processed_options = []
       end
 
+      def processed_options=(processed_options)
+        @processed_options.concat(processed_options)
+        @args_parsed = true
```

Comment:
I'd suggest putting this last just to avoid it being set if e.g. the `freeze` fails.

Reply:
Done.